flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveSchedulerTest.java

```diff
@@ -119,6 +123,21 @@
     public static final TestExecutorResource<ScheduledExecutorService> TEST_EXECUTOR_RESOURCE =
             new TestExecutorResource<>(Executors::newSingleThreadScheduledExecutor);
 
+    @Parameterized.Parameters(name = "execution mode = {0}")
+    public static Object[] data() {
+        return new Object[] {SchedulerExecutionMode.REACTIVE, null};
```

Comment:
I do wonder whether we really need to run all tests with reactive mode, or have one targeted tests to test that the right VertexParallelismStore is computed

Reply:
I considered that as well, though the core logic for requesting resources and submitting graphs now depends on quite a bit on execution mode. I'll review and see if there are cases where this can be reduced.

Reply:
I think it is enough to add the just targetted tests that you are talking about, though I'd also like to add one to ensure that gets carried through to the resource requests.

Reply:
I'll also break apart the `AdaptiveScheduler#createExecutionGraphWithAvailableResourcesAsync` method to test the adjusted job graph logic.